src/components/Nav.tsx

```diff
@@ -0,0 +1,38 @@
+"use client";
+
+import { useState } from "react";
+import { NavButton } from "./NavButton";
+import { usePathname } from "next/navigation";
+
+type NavProps = Readonly<{
+  navButtons: Array<{ id: number; href: string; content: React.JSX.Element }>;
+}> &
+  React.ComponentPropsWithRef<"nav">;
+
+export const Nav = ({ className, navButtons }: NavProps) => {
+  const pathname = usePathname();
+  const defaultCurrent =
+    navButtons.find((navButton) => pathname.includes(navButton.href))?.id || 0;
+  const [activeNav, setActiveNav] = useState(defaultCurrent);
```

Comment:
I think `useState` is not mandatory here since usePathname always returns the current displayed page.
I wonder if we can't simplify this by adding an "active" attribute to the NavButton (since it's actually a specific state for the button) and use it like this :

```
export const Nav = ({ className, navButtons }: NavProps) => {
  const pathname = usePathname();
  return (
    <nav className={className}>
      {navButtons.map(({ id, href, content }) => (
        <NavButton active={pathname.includes(href)} key={id} href={href}>
          {content}
        </NavButton>
      ))}
    </nav>
  );
};
```

```
type NavButtonProps = Readonly<{
  active?: boolean;
  children: ReactNode;
  href: string;
  onClick?: MouseEventHandler | undefined;
}> &
  React.ComponentProps<"button">;

export const NavButton = ({
  active = false,
  children,
  href,
  className,
}: NavButtonProps) => (
  <Link
    href={href}
    className={clsx(
      "rounded-tl-lg rounded-bl-lg px-4 py-2 text-center text-white bg-secondary hover:scale-up-center",
      { "bg-tertiary": active },
      className,
    )}
  >
    {children}
  </Link>
);
```

What do you think ?